src/abstractdataframe/show.jl

```diff
@@ -278,13 +278,28 @@ function _show(io::IO,
     # case, we will add this information using the row name column of
     # PrettyTables.jl. Otherwise, we can just use the row number column.
     if (rowid === nothing) || (ncol(df) == 0)
-        show_row_number = true
+        show_row_number::Bool = haskey(kwargs, :show_row_number) ? kwargs[:show_row_number] : true
         row_names = nothing
+
+        # If the columns with row numbers is not shown, then we should not
+        # display a vertical line after the first column.
+        vlines = show_row_number ? [1] : Int[]
```

Comment:
it occurred to me that we can write it as 😄 (but if you prefer your line I am OK with it also):
```suggestion
        vlines = fill(1, show_row_number)
```